test/extended/pod_autoscaling/deployment.go

```diff
@@ -0,0 +1,148 @@
+package pod_autoscaling
+
+import (
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	exapi "k8s.io/kubernetes/pkg/apis/extensions"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	"k8s.io/kubernetes/pkg/util/wait"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+)
+
+var _ = g.Describe("Horizontal Pod Autoscaling: DeploymentConfigs", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		oc = exutil.NewCLI("hpa-tests", exutil.KubeConfigPath())
+		deploymentConfigFixture = exutil.FixturePath("fixtures", "test-basic-deploymentconfig.yaml")
+		dcName = "frontend"
+	)
+
+	g.JustBeforeEach(func() {
+		g.By("creating the DeploymentConfig")
+		err := oc.Run("create").Args("-f", deploymentConfigFixture).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	g.It("should scale the deployment when it is complete", func() {
+		err := oc.Run("deploy").Args(dcName, "--latest").Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		rcClient := oc.KubeREST().ReplicationControllers(oc.Namespace())
+		err = exutil.WaitForADeployment(rcClient, dcName, exutil.CheckDeploymentCompletedFunc, exutil.CheckDeploymentFailedFunc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("creating a new HPA targeted at 80% CPU consumption")
+		scaler := exapi.HorizontalPodAutoscaler{
+			ObjectMeta: kapi.ObjectMeta{Name: "frontend-scaler"},
+			Spec: exapi.HorizontalPodAutoscalerSpec{
+				ScaleRef: exapi.SubresourceReference{
+					Kind: "DeploymentConfig",
+					Name: dcName,
+					APIVersion: "v1",
+					Subresource: "scale",
+				},
+				MaxReplicas: 6,
+				CPUUtilization: &exapi.CPUTargetUtilization{TargetPercentage: 80},
+			},
+		}
+		_, err = oc.KubeREST().Extensions().HorizontalPodAutoscalers(oc.Namespace()).Create(&scaler)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		err = waitForRCToBeScaled(rcClient, dcName+"-1", 1)
```

Comment:
ah right, thanks for the explanation

On Mon, Nov 16, 2015 at 8:25 PM, Solly Ross <notifications@github.com>
wrote:

> In test/extended/pod_autoscaling/deployment.go
> <https://github.com/openshift/origin/pull/5879#discussion_r44969359>:
>
> > +			ObjectMeta: kapi.ObjectMeta{Name: "frontend-scaler"},
> > +			Spec: exapi.HorizontalPodAutoscalerSpec{
> > +				ScaleRef: exapi.SubresourceReference{
> > +					Kind: "DeploymentConfig",
> > +					Name: dcName,
> > +					APIVersion: "v1",
> > +					Subresource: "scale",
> > +				},
> > +				MaxReplicas: 6,
> > +				CPUUtilization: &exapi.CPUTargetUtilization{TargetPercentage: 80},
> > +			},
> > +		}
> > +		_, err = oc.KubeREST().Extensions().HorizontalPodAutoscalers(oc.Namespace()).Create(&scaler)
> > +		o.Expect(err).NotTo(o.HaveOccurred())
> > +
> > +		err = waitForRCToBeScaled(rcClient, dcName+"-1", 1)
>
> Why are you targeting for one replica? Dc spec says 3 and the autoscaler
> is setup to max 6.
>
> The scale should scale it down to 1 because its using a tiny amount of CPU
> (the request is 500 millicores * 80% = 400 millicores, and the image uses >
> 10 millicores when idle)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/5879/files#r44969359>.
>